src/lib.rs

```diff
@@ -173,8 +174,28 @@ fn init_metrics(config: Config) -> (Option<PrometheusRegistry>, SdkMeterProvider
                     Box::new(DefaultTemporalitySelector::new())
                 };
 
-            let exporter = match opentelemetry_otlp::new_exporter()
-                .tonic()
+            let mut exporter_builder = opentelemetry_otlp::new_exporter().tonic();
+            if export_target.url.starts_with("https") || export_target.url.starts_with("grpcs") {
+                if let Some(ca_cert_path) = export_target.ca_cert_path {
+                    let ca_cert = std::fs::read(ca_cert_path);
+                    match ca_cert {
+                        Ok(ca_cert) => {
+                            let ca_cert = Certificate::from_pem(ca_cert);
+                            let tls_config = ClientTlsConfig::new().ca_certificate(ca_cert);
+                            exporter_builder = exporter_builder.with_tls_config(tls_config);
+                        }
+                        Err(e) => {
+                            error!("unable to load ca_cert_file {:?}", e);
+                            continue;
+                        }
+                    }
+                }
+            } else {
+                exporter_builder =
+                    exporter_builder.with_tls_config(ClientTlsConfig::new().with_native_roots());
+            }
```

Comment:
To avoid duplicating this code in logger.rs, consider moving this to a function that can be called from both locations (e.g. `create_exporter_builder(ca_cert_path: Option<String>) -> ExportConfig`)

Reply:
Good call out. Done.